File: ScaleformUI_Csharp/Menus/Pause Menus/PauseMenu/Items/PauseMenuItem.cs
using ScaleformUI.Elements;
using ScaleformUI.Menus;

namespace ScaleformUI.PauseMenu
{
    public class PauseMenuItem
    {
        public ItemFont LabelFont = ScaleformFonts.CHALET_LONDON_NINETEENSIXTY;
        public ScaleformLabel Label { get; set; }
        public TabLeftItem ParentLeftItem { get; set; }
        public BaseTab ParentTab { get; set; }
        public PM_Column ParentColumn { get; set; }
        public PauseMenuItem(string label)
        {
            Label = label;
        }
        public PauseMenuItem(string label, ItemFont labelFont)
        {
            Label = label;
            LabelFont = labelFont;
        }

        public virtual bool Selected { get; set; }
    }
}

File: ScaleformUI_Lua/src/Menus/PauseMenus/PauseMenu/Items/_PauseMenuItem.lua
PauseMenuItem = setmetatable({}, PauseMenuItem)
PauseMenuItem.__index = PauseMenuItem
PauseMenuItem.__call = function()
    return "PauseMenuItem", "PauseMenuItem"
end

---@class PauseMenuItem
---@field Label string
---@field Parent PauseMenuItem

---Creates a new PauseMenuItem.
---@param label string
---@return PauseMenuItem
function PauseMenuItem.New(label, labelFont)
    local data = {
        label = label or "",
        LabelFont = labelFont or ScaleformFonts.CHALET_LONDON_NINETEENSIXTY,
        labelFont = labelFont or ScaleformFonts.CHALET_LONDON_NINETEENSIXTY,
        selected = false,
        ParentTab = nil,
        PrentColumn = nil,
    }
    return setmetatable(data, PauseMenuItem)
end

function PauseMenuItem:Selected(bool)
    if bool == nil then
        return self.selected
    else
        self.selected = bool
    end
end
